services/storage/virtualharddisk/client.go

```diff
@@ -72,6 +74,33 @@ func (c *VirtualHardDiskClient) Resize(ctx context.Context, group, container, na
 	return err
 }
 
+// Upload methods invokes upload of the storage resource to target sasurl
+func (c *VirtualHardDiskClient) Upload(ctx context.Context, group, container, name string, targetUrl string) error {
+	fmt.Printf("moc-sdk-for-go: client.go: Upload start: vhd name [%s] to [%s]\n", name, targetUrl)
+	fmt.Printf("moc-sdk-for-go: client.go: Calling Get on VHD name\n")
+
+	vhds, err := c.Get(ctx, group, container, name)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("moc-sdk-for-go: client.go: Done with Get\n")
+
+	if vhds == nil || len(*vhds) == 0 {
+		return errors.Wrapf(errors.NotFound, "%s", name)
+	}
+	fmt.Printf("moc-sdk-for-go: client.go: Found vhd as len is not zero\n")
+
+	if targetUrl == "" {
+		return errors.Wrapf(errors.InvalidInput, "targetUrl cannot be empty")
+	}
+	fmt.Printf("moc-sdk-for-go: client.go: Targeturl is not empty\n")
+
+	vhd := (*vhds)[0]
```

Comment:
Add a comment or error handling for cases where we have more than 1 vhd

Reply:
Added